### PR TITLE
usman/Edit Payment Methods in P2P - Responsive

### PR DESCRIPTION
**Pull request by usman-deriv**
## Description
This pull request is for automation for updating already existing payment methods for P2P in responsive mode. 

### Implementations
- Improved readability by using descriptive variable names.
- Optimized code structure. Used functions to improve code organization and reusability. 
- Regardless of payment methods saved in the user's profile, the code is designed to handle all payment methods. i.e. Bank Transfers, E-wallets, and Others. 

### ClickUp Card
[Create Cypress test QATEST-2831 My Profile page - Edit Payment Method](https://app.clickup.com/t/20696747/P2PS-2449)

### CODE REVIEW

File: cypress/e2e/wip/P2P/editPaymentMethods.cy.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        let notification = doc.querySelector('.notification__header')
+        if (notification && notification != undefined && notification != null) {
+            cy.log('Notification header appeared')
+            cy.get('.notification__text-body').invoke('text').then((text) => {
+                cy.log(text)
+            })
+            cy.get('.notification__close-button').should('be.visible').click()
+            cy.get('.notification__header').should('not.exist')
+            notification = null;
+            cy.wait(2000)
```
Comment (markodonnell-deriv) [inline cypress/e2e/wip/P2P/editPaymentMethods.cy.js:28]: Is this wait necessary, we try and avoid these in Cypress as waits are built in to functions such as click and should?

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Good first test. 
I notice there aren't any testing library commands in this test. If you're finding it difficult to identify them, use the Playwright Locator Generator as per the e2e docs: https://wikijs.deriv.cloud/en/Quality-Assurance/e2e-test-automation

File: cypress/e2e/wip/P2P/editPaymentMethods.cy.js (hunk 1 above)
Comment (usman-deriv) [reply cypress/e2e/wip/P2P/editPaymentMethods.cy.js:28]: This function checks and dynamically closes notifications, regardless of their number (of notifications). In **responsive**, the notifications come one by one and do not list down as they list on the full-screen web. 

When 1 notification is closed, the function is called again recursively to check for the next notification (if exists), as the first one is still closing, the cypress gets the objects of the first one as it still exists (the notification fades and closes slowly). The wait is just to wait for the notification to get removed from the dom.

I have tried to run tests without using wait but it then tries to get the 'x' button for the already closed notification and it fails there, hence this adequate wait is necessary.

Comment (usman-deriv) [reply cypress/e2e/wip/P2P/editPaymentMethods.cy.js]: [Update] - This wait command has been removed after debugging.

File: cypress/e2e/wip/P2P/editPaymentMethods.cy.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            cy.get('.dc-checkbox__box').should('be.visible').click()
+        }
+    })
+    cy.findByRole('button', { name: 'Confirm' }).should('be.visible').click()
+}
+
+function closeNotificationHeader() {
+    cy.document().then(doc => {
+        let notification = doc.querySelector('.notification__header')
+        if (notification && notification != undefined && notification != null) {
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/P2P/editPaymentMethods.cy.js:20]: Won't this work with just `if(notification)`?

Comment (usman-deriv) [reply cypress/e2e/wip/P2P/editPaymentMethods.cy.js]: This has been updated, thank you.

File: cypress/e2e/wip/P2P/editPaymentMethods.cy.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+}
+
+describe("QATEST-2831 - My Profile page - Edit Payment Method", () => {
+    beforeEach(() => {
+        cy.c_login()
+        cy.c_visitResponsive('/appstore/traders-hub', 'small')
+    })
+
+    it('should be able to edit the existing payment methods - responsive', () => {
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/P2P/editPaymentMethods.cy.js:93]: Sorry 😅 Just one more thing. Can you please capitalize the start of this sentence
image

Comment (usman-deriv) [reply cypress/e2e/wip/P2P/editPaymentMethods.cy.js]: This is done and thank you! :)

**Review by abuzar-deriv — APPROVED** [review]
Looks good.

File: cypress/e2e/wip/P2P/editPaymentMethods.cy.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+function closeNotificationHeader() {
+    cy.document().then(doc => {
+        let notification = doc.querySelector('.notification__header')
+        if (notification) {
+            cy.log('Notification header appeared')
+            cy.get('.notification__text-body').invoke('text').then((text) => {
+                cy.log(text)
+            })
+            cy.findByRole('button', { name: 'Close' }).should('be.visible').click().and('not.exist')
+            notification = null;
```
Comment (markodonnell-deriv) [inline cypress/e2e/wip/P2P/editPaymentMethods.cy.js:26]: remove ;

File: cypress/e2e/wip/P2P/editPaymentMethods.cy.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+function navigateToTab(tabName) {
+    cy.findByText(tabName).click()
+}
+
+function setText(fieldName, fieldText) {
+    cy.findByRole('textbox', { name: fieldName }).clear().type(fieldText).should('have.value', fieldText)
+}
+
+function generateAccountNumberString(length) {
+    let result = '';
```
Comment (markodonnell-deriv) [inline cypress/e2e/wip/P2P/editPaymentMethods.cy.js:50]: remove ;

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Good use of testing library now but please remove semi-colon as per coding standards in this doc: https://wikijs.deriv.cloud/en/Quality-Assurance/e2e-test-automation

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Pull request merged**